controllers/wavefrontalert_controller.go

```diff
@@ -105,39 +133,123 @@ func (r *WavefrontAlertReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+			wfAlert.Status = alertmanagerv1alpha1.WavefrontAlertStatus{
+				RetryCount:         wfAlert.Status.RetryCount + 1,
+				ErrorDescription:   err.Error(),
+				State:              state,
+				LastChangeChecksum: lastChangeChecksum,
+			}
+			return r.CommonClient.UpdateStatus(ctx, &wfAlert, state, errRequeueTime)
+		}
+
+		status.State = alertmanagerv1alpha1.Ready
+		alertResponse := alertmanagerv1alpha1.Alert{
+			ID:                 *alert.ID,
+			Name:               alert.Name,
+			Link:               fmt.Sprintf("https://%s/alerts/%s", config.Props.WavefrontAPIUrl(), *alert.ID),
+			LastChangeChecksum: lastChangeChecksum,
+		}
+		status.Alerts = []alertmanagerv1alpha1.Alert{alertResponse}
+		status.RetryCount = 0
+		wfAlert.Status = status
+
+		return r.CommonClient.UpdateStatus(ctx, &wfAlert, alertmanagerv1alpha1.Ready)
 	}
 
-	// Call wavefront apis and create/update the alerts
+	// existing alert - Perform the updateAlert one by one
 
-	// look for the status and call wavefront APIs to delete alert/s
-	// if its create/update
-	// if exportedParams is empty, proceed with wavefront apis to create the alert
-	// exportedParams is not empty and no status available- This is considered as brand new request
-	// don't do anything
-	// if there is a diff in exportedParams checksum compared to status
-	// don't do anything
-	// else
-	// look for the status and proceed with wavefront apis to update the alert
+	status.State = wfAlert.Status.State
+	var respAlerts []alertmanagerv1alpha1.Alert
+	for _, a := range wfAlert.Status.Alerts {
+		alert := wf.Alert{
+			ID: &a.ID,
+		}
+		r.convertAlertCR(ctx, &wfAlert, &alert)
+		state := alertmanagerv1alpha1.Ready
+		respAlert := alertmanagerv1alpha1.Alert{
+			ID:                 a.ID,
+			Name:               a.Name,
+			LastChangeChecksum: lastChangeChecksum,
+			Link:               fmt.Sprintf("https://%s/alerts/%s", config.Props.WavefrontAPIUrl(), *alert.ID),
+		}
+		// TODO: Only do the UpdateAlert if there is a difference between parent lastChangeChecksum and child lastChangeChecksum- This could be in a scenario
+		//  where it updated 99 out of 100 child alerts and 1 got failed and it got requeued. so instead of trying to update 100 again lets just do only 1 api
+		// call update api
+		if err := r.WavefrontClient.UpdateAlert(ctx, &alert); err != nil {
+			r.Recorder.Event(&wfAlert, v1.EventTypeWarning, err.Error(), "unable to update the alert")
+			state = alertmanagerv1alpha1.Error
+			if strings.Contains(err.Error(), "Exceeded limit setting") {
+				// For ex: error is "Exceeded limit setting: 100 alerts allowed per customer"
+				state = alertmanagerv1alpha1.ClientExceededLimit
+			}
+			log.Error(err, "unable to create the alert")
+			respAlert.State = state
+			respAlert.LastChangeChecksum = a.LastChangeChecksum
+			// if even one of the child got failed, make parent status as error
+			status.State = state
+			status.RetryCount = wfAlert.Status.RetryCount
```

Comment:
We should return here since it is a error case. Also adding retry and requeue?

Reply:
Not necessarily. If you look carefully, its under the loop for multiple alerts situation. what it is doing is, it is just updating every individual alert state. And if one of the child status is error it makes the parent status error which automatically qualified for retry by controller Kubebuilder implementation

Reply:
okay, so error state will be requeue immediately due to this [code](https://github.com/keikoproj/alert-manager/blob/fa580cb4e7a6651f26a311026daf10e9780085b3/controllers/common/common.go#L82) , should we increase the retry count in parent status?